src/wasm-bindgen.rs

```diff
@@ -10,11 +10,21 @@ use crate::Error;
 extern crate std;
 use std::thread_local;
 
+use js_sys::Uint8Array;
 use wasm_bindgen::prelude::*;
 
+// Maximum is 65536 bytes see https://developer.mozilla.org/en-US/docs/Web/API/Crypto/getRandomValues
+const BROWSER_CRYPTO_BUFFER_SIZE: usize = 256;
+
+struct BrowserCryptoContext {
+    crypto: BrowserCrypto,
+    // A temporary buffer backed by browser memory to avoid multithreaded wasm exception. See issue #164
+    buf: Uint8Array,
+}
+
```

Comment:
Instead of having this large structure, we could just do:

```rust
enum RngSource {
    Node(NodeCrypto),
    Browser(BrowserCrypto, Uint8Array),
}
```

and then add the comment about using the temporary buffer  when we actually use the temp buffer:

```rust
RngSource::Browser(crypto, buf) => {
    // getRandomValues does not work with all types of WASM memory, so
    // we initially write to browser memory to avoid an exception.
    for chunk in dest.chunks_mut(BROWSER_CRYPTO_BUFFER_SIZE) {
        ...
    }
}
```

Reply:
Sorry for the late response, but do you still want me to fix this? Since you force pushed to my branch and already approved it

Reply:
Yes, fixing this would be preferable (I can also do it when i have time).